src/io/io.jl

```diff
@@ -1,2 +1,141 @@
 include("write_vtk.jl")
 include("read_vtk.jl")
+
+function write_meta_data(callback::Union{SolutionSavingCallback, PostprocessCallback},
+                         integrator)
+    semi = integrator.p
+    (; systems) = semi
+
+    output_directory = callback.output_directory
+    prefix = callback.prefix
+    git_hash = callback.git_hash
+
+    filenames = system_names(systems)
+
+    foreach_system(semi) do system
+        system_index = system_indices(system, semi)
+
+        write_meta_data(system; system_name=filenames[system_index], output_directory,
+                        prefix,
+                        git_hash)
+    end
+end
+
+function write_meta_data(system; system_name, output_directory, prefix, git_hash)
+    mkpath(output_directory)
+
+    meta_data = Dict{String, Any}(
+        "solver_name" => "TrixiParticles.jl",
+        "solver_version" => git_hash,
+        "julia_version" => string(VERSION)
+    )
+
+    get_meta_data!(meta_data, system)
```

Comment:
```suggestion
    add_meta_data!(meta_data, system)
```
IMO this is a better function name